src/java/org/apache/cassandra/tcm/sequences/BootstrapAndJoin.java

```diff
@@ -217,6 +218,8 @@ public SequenceState executeNext()
                                         "For more, see `nodetool help bootstrap`. {}", SystemKeyspace.getBootstrapState());
                             return halted();
                         }
+                        // this node might have just bootstrapped; check if we should run repair immediately
+                        AutoRepairUtils.runRepairOnNewlyBootstrappedNodeIfEnabled();
```

Comment:
Maybe paranoid on my part, but could we move this towards the end, maybe in the end of the `if (finishJoiningRing)` block?  This way is something unexpected happens in this code path, it doesn't skip marking the bootstrap as complete/updating CMS.

Reply:
Same applies for `BootstrapAndReplace`/`ReplaceSameAddress`